### DISCUSSION TITLE
Internal Energy Value to a Postprocessor?

### DISCUSSION THREAD

**Discussion by Vandenbg** [idaholab/moose#25843] [Q&A General] [answered]
Hello, I am looking for a way to output the internal energy to a post processor. This kernel calculates the internal energy but I am not sure how to connect this with a post processor.
```
[kernels]
  [heat_ie]
    type = HeatConductionTimeDerivative
    variable = temperature
  []
...
```
Internal energy is not a material property or variable. I thought it would be relatively simple, however, I haven't found a solution. Any help would be appreciated.

**Comment by GiudGiud** [comment, marked as answer]
Hello

Internal energy depends on temperature so it s more of a field than a single value. 

Do you want to integrate it over a subdomain?

Guillaume

**Reply by Vandenbg** [reply]
Yes that would be ideal.

**Reply by GiudGiud** [reply]
are rho and cp constant or are they temperature-dependent (so kind of fields even if you have not defined them) as well?

**Reply by Vandenbg** [reply]
They are temperature dependent, I haven't explicitly defined them. I can output them as material properties.

**Reply by GiudGiud** [reply]
You have two options:
make a material that defines the internal energy material property and output that
or
use MaterialRealAux to output rho and cp to auxiliary variable then a ParsedAux to multiply all those. 
The second one will be more approximate (averages rho, cp to elements instead of integrating rho cp T directly)

**Reply by GiudGiud** [reply]
this has been done for HeatStructure components in the TH module but you are not using that

**Reply by Vandenbg** [reply]
Thank you, I'll give this a try. I will look at the TH module for reference.
